modules/core/src/test/kotlin/datamaintain/core/step/sort/ByCaseInsensitiveSeparatorFreeAlphabeticalSortingStrategyTest.kt

```diff
@@ -0,0 +1,155 @@
+        val script3 = ExecutedScript("v1.1.0-1", "checksum3", "1.1.0-1", ExecutionStatus.OK)
+        val script4 = ExecutedScript("v1.2", "checksum4", "1.2", ExecutionStatus.OK)
+        val script5 = ExecutedScript("v11", "checksum5", "11", ExecutionStatus.OK)
+        val script6 = ExecutedScript("v101", "checksum6", "101", ExecutionStatus.OK)
+
+        // When
+        expectThat(sortingStrategy.sort(listOf(script4, script1, script5, script3, script6, script2), Script::identifier)) {
+            // Then
+            size.isEqualTo(6)
+            first().isEqualTo(script1)
+            get(1).isEqualTo(script2)
+            get(2).isEqualTo(script3)
+            get(3).isEqualTo(script4)
+            get(4).isEqualTo(script5)
+            last().isEqualTo(script6)
+        }
+    }
+
+    @Test
+    fun `should sort scripts list by complex identifier with only char`() {
+        // Given
+        val script1 = ExecutedScript("va", "checksum1", "a", ExecutionStatus.OK)
+        val script2 = ExecutedScript("va.a", "checksum2", "a.a", ExecutionStatus.OK)
+        val script3 = ExecutedScript("va.a.b-a", "checksum3", "a.a.b-a", ExecutionStatus.OK)
+        val script4 = ExecutedScript("va.b", "checksum4", "a.b", ExecutionStatus.OK)
+        val script5 = ExecutedScript("vaa", "checksum5", "aa", ExecutionStatus.OK)
+
+        // When
+        expectThat(sortingStrategy.sort(listOf(script4, script1, script5, script3, script2), Script::identifier)) {
+            // Then
+            size.isEqualTo(5)
+            first().isEqualTo(script1)
+            get(1).isEqualTo(script2)
+            get(2).isEqualTo(script3)
+            get(3).isEqualTo(script4)
+            last().isEqualTo(script5)
+        }
+    }
+
+    @Test
+    fun `should sort scripts list by complex identifier`() {
+        // Given
+        val script1 = ExecutedScript("va", "checksum1", "a", ExecutionStatus.OK)
+        val script2 = ExecutedScript("va.1", "checksum2", "a.1", ExecutionStatus.OK)
+        val script3 = ExecutedScript("va.1.b-1", "checksum3", "a.1.b-1", ExecutionStatus.OK)
+        val script4 = ExecutedScript("va.1.b-2", "checksum4", "a.1.b-2", ExecutionStatus.OK)
+        val script5 = ExecutedScript("va.1.c", "checksum5", "a.1.c", ExecutionStatus.OK)
+        val script6 = ExecutedScript("va.2", "checksum6", "a.2", ExecutionStatus.OK)
+        val script7 = ExecutedScript("vaa", "checksum7", "aa", ExecutionStatus.OK)
+
+        // When
+        expectThat(sortingStrategy.sort(listOf(script4, script6, script1, script5, script3, script7, script2), Script::identifier)) {
+            // Then
+            size.isEqualTo(7)
+            first().isEqualTo(script1)
+            get(1).isEqualTo(script2)
+            get(2).isEqualTo(script3)
+            get(3).isEqualTo(script4)
+            get(4).isEqualTo(script5)
+            get(5).isEqualTo(script6)
+            last().isEqualTo(script7)
+        }
+    }
+}
```

Comment:
I think your tests lack an example with an identifier containing more values after splits but still be before shorter identifiers. An example could be scrips with identifiers "v0.1" and "v1".

Reply:
Not sure to understand the test cas you give me, but understand I forget an important test case : "abc" and "bc". The current algorithm will sort ["bc", "abc"] which is not what we want.
But if do the test with "111" and "11", we want ["11", "111"].
Mixing figure and letter is not so simple. I will try to fix that, without complefixying too much the algorithm.